sphinx/config.py

```diff
@@ -139,6 +139,9 @@ class Config(object):
         numfig_secnum_depth = (1, 'env', []),
         numfig_format = ({}, 'env', []),  # will be initialized in init_numfig_format()
 
+        math_number_all = (False, 'env', []),
+        math_eqref_format = (None, 'env', string_classes),
+        math_numfig = (True, 'env', []),
```

Comment:
We have to discuss about integration of `math_eqref_format` with `numref_format`, and `math_numfig` with `numfig`.

Reply:
I meant the goal of this PR is integration maths to sphinx-core. That is like a refactoring. So integration of config-entries are off topic of this PR.
So let's discuss in another issue.

Reply:
I made #4991.